### PR TITLE
trash-cli 0.17.1.14 (new formula)

### PR DESCRIPTION
**Pull request by jdanford**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This is a formula for [trash-cli](https://github.com/andreafrancia/trash-cli), a command line interface to the freedesktop.org trashcan.

### CODE REVIEW

**Comment by jdanford** [comment]
Apologies for repeatedly failing CI! For some reason `brew audit --online --new-formula trash-cli` didn't output anything for me, which I assumed meant it passed. Any idea what the problem is?

**Comment by ilovezfs** [comment]
@jdanford what does `brew config` say for you locally?

**Comment by ilovezfs** [comment]
Also, `brew doctor`

**Comment by jdanford** [comment]
`brew config`:
```
HOMEBREW_VERSION: 1.3.1-40-ge1bab7e
ORIGIN: https://github.com/Homebrew/brew
HEAD: e1bab7ede9461a6d5c93fb7240ba5a1f3bf7a7fb
Last commit: 19 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: ab72fee8166749231036a4476aaba30df9619800
Core tap last commit: 11 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: octa-core 64-bit haswell
Homebrew Ruby: 2.0.0-p648
Clang: 8.1 build 802
Git: 2.14.1 => /usr/local/bin/git
Perl: /usr/local/bin/perl => /usr/local/Cellar/perl/5.26.0/bin/perl
Python: /usr/local/opt/python/libexec/bin/python => /usr/local/Cellar/python/2.7.13_1/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /usr/local/bin/ruby => /usr/local/Cellar/ruby/2.4.1_1/bin/ruby
Java: 1.8.0_131
macOS: 10.12.6-x86_64
Xcode: N/A
CLT: 8.3.2.0.1.1492020469
X11: N/A
```
----
`brew doctor`:
```
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry and just ignore them. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and what additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew provided
script of the same name. We found the following "config" scripts:
  /Users/jsd/.cargo/bin/cargo-install-update-config
  /Users/jsd/.pyenv/shims/python-config
  /Users/jsd/.pyenv/shims/python2-config
  /Users/jsd/.pyenv/shims/python2.7-config
  /Users/jsd/.pyenv/shims/python3-config
  /Users/jsd/.pyenv/shims/python3.4-config
  /Users/jsd/.pyenv/shims/python3.4m-config
  /Users/jsd/.pyenv/shims/python3.6-config
  /Users/jsd/.pyenv/shims/python3.6m-config

Warning: Putting non-prefixed coreutils in your path can cause gmp builds to fail.

Warning: Putting non-prefixed findutils in your path can cause python builds to fail.
```

I tried uninstalling `coreutils` and `findutils`, but `brew audit --new-formula trash-cli` still gives me no output and a return code of `0`.

**Comment by ilovezfs** [comment]
Well it looks like it's passing now, so that would be expected right?

**Comment by ilovezfs** [comment]
Can you give an example where you think it should print a message and it's not?

**Comment by jdanford** [comment]
I changed the indentation to four spaces and installed it locally, so shouldn't I expect a message like `Use 2 (not 4) spaces for indentation.`?

**Comment by jdanford** [comment]
Doesn't seem like my `brew audit` problem has an obvious solution, but I'll keep looking. Anyway, this PR is no longer failing, so is it ready to be merged?

File: Formula/trash-cli.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  depends_on :python if MacOS.version <= :snow_leopard
+
+  conflicts_with "trash", :because => "both install a `trash` binary"
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    assert_match "trash #{version}", shell_output("#{bin}/trash --version")
```
Comment (dunn) [inline Formula/trash-cli.rb]: Can you `touch` a temporary file then try trashing it?  Also do `#{bin}/trash` just to ensure the path is right.

Comment (ilovezfs) [inline Formula/trash-cli.rb]: This needs to be a test that exercises and verifies the functionality of the software once installed, such as the test suggested by @dunn.

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/trash-cli.rb (hunk 1 above)
Comment (jdanford) [inline Formula/trash-cli.rb]: > Also do `#{bin}/trash`

Can you elaborate on what you mean?

Comment (ilovezfs) [inline Formula/trash-cli.rb]: The test needs to verify the trashing behavior actually works.

Comment (dunn) [inline Formula/trash-cli.rb]: @jdanford I mean call `bin/"trash"` instead of just `"trash"` so make sure the correct binary is executed.

Comment (jdanford) [inline Formula/trash-cli.rb]: I'm already doing so in the `shell_output` call on the right – the `"trash #{version}"` string on the left is the expected output. Anyway, should I include both the version test and the temporary file test, or just the latter?

Comment (ilovezfs) [inline Formula/trash-cli.rb]: Just the latter.

Comment (jdanford) [inline Formula/trash-cli.rb]: Done!

**Comment by jdanford** [comment]
Forgot to update the SHA, will fix later today!

**Comment by fxcoudert** [comment]
Thanks @jdanford for the pull request!

**Pull request closed without merging**